starter-core/src/test/groovy/io/micronaut/starter/feature/spring/SpringDataJdbcSpec.groovy

```diff
@@ -53,118 +55,21 @@ class SpringDataJdbcSpec extends ApplicationContextSpec {
     }
 
     @Unroll
-    void 'test spring-data-jdbc with Gradle for language=#language'() {
+    void 'test spring-data-jdbc with buildTool=#buildTool for language=#language'(Language language, BuildTool buildTool) {
         when:
-        String template = new BuildBuilder(beanContext, BuildTool.GRADLE)
-                .features(['spring-data-jdbc', 'kapt'])
+        String template = new BuildBuilder(beanContext, buildTool)
+                .features(['spring-data-jdbc'])
                 .language(language)
                 .render()
 
-        then:
-        template.contains("${getGradleAnnotationProcessorScope(language)}(\"io.micronaut.spring:micronaut-spring-annotation\")")
-        template.contains('implementation("io.micronaut.data:micronaut-data-spring")')
-        template.contains('implementation("org.springframework:spring-jdbc")')
-
-        where:
-        language << Language.values().toList()
-    }
-
-    void 'test maven spring-data-jdbc feature for java'() {
-        when:
-        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
-                .features(['spring-data-jdbc'])
-                .language(Language.JAVA)
-                .render()
+        BuildTestVerifier verifier = BuildTestUtil.verifier(buildTool, language, template)
 
         then:
-        template.contains("""
-    <dependency>
-      <groupId>io.micronaut.data</groupId>
-      <artifactId>micronaut-data-spring</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-    <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-jdbc</artifactId>
-      <scope>compile</scope>
-    </dependency>
-""")
-        template.contains("""
-            <path>
-              <groupId>io.micronaut.spring</groupId>
-              <artifactId>micronaut-spring-annotation</artifactId>
-              <version>\${micronaut.spring.version}</version>
-              <exclusions>
-                <exclusion>
-                  <groupId>io.micronaut</groupId>
-                  <artifactId>micronaut-core</artifactId>
-                </exclusion>
-              </exclusions>
```

Comment:
We used to test that this exclusion existed for Maven

We no longer seem to do that?

Reply:
I will add that back.

Reply:
ping @timyates this exposed an issue with the hasExclusions API (didn't work for Maven annotation processors). Now fixed.